tests/TestCase/Mailer/MailerTest.php

```diff
@@ -14,85 +14,383 @@
+
+        $result = $this->mailer->getSubject();
+        $this->assertEquals($configs['subject'], $result);
+
+        $result = $this->mailer->getTransport();
+        $this->assertInstanceOf(DebugTransport::class, $result);
+
+        $result = $this->mailer->deliver('This is the message');
+
+        $this->assertStringContainsString('Message-ID: ', $result['headers']);
+        $this->assertStringContainsString('To: ', $result['headers']);
+    }
+
+    /**
+     * testConfigArrayWithLayoutWithoutTemplate method
+     *
+     * @return void
+     */
+    public function testConfigArrayWithLayoutWithoutTemplate()
+    {
+        $configs = [
+            'from' => ['some@example.com' => 'My website'],
+            'to' => 'test@example.com',
+            'subject' => 'Test mail subject',
+            'transport' => 'debug',
+            'layout' => 'custom',
+        ];
+        $this->mailer = new Mailer($configs);
+
+        $template = $this->mailer->viewBuilder()->getTemplate();
+        $layout = $this->mailer->viewBuilder()->getLayout();
+        $this->assertNull($template);
+        $this->assertEquals($configs['layout'], $layout);
+    }
+
+    /**
+     * testConstructWithConfigString method
+     *
+     * @return void
+     */
+    public function testConstructWithConfigString()
+    {
+        $configs = [
+            'from' => ['some@example.com' => 'My website'],
+            'to' => 'test@example.com',
+            'subject' => 'Test mail subject',
+            'transport' => 'debug',
+        ];
+        Mailer::setConfig('test', $configs);
+
+        $this->mailer = new Mailer('test');
+
+        $result = $this->mailer->getTo();
+        $this->assertEquals([$configs['to'] => $configs['to']], $result);
+
+        $result = $this->mailer->getFrom();
+        $this->assertEquals($configs['from'], $result);
+
+        $result = $this->mailer->getSubject();
+        $this->assertEquals($configs['subject'], $result);
+
+        $result = $this->mailer->getTransport();
+        $this->assertInstanceOf('Cake\Mailer\Transport\DebugTransport', $result);
+
+        $result = $this->mailer->deliver('This is the message');
+
+        $this->assertStringContainsString('Message-ID: ', $result['headers']);
+        $this->assertStringContainsString('To: ', $result['headers']);
+    }
+
+    /**
+     * test profile method
+     *
+     * @return void
+     */
+    public function testSetProfile()
+    {
+        $config = ['to' => 'foo@bar.com'];
+        $this->mailer->setProfile($config);
+        $this->assertSame(['foo@bar.com' => 'foo@bar.com'], $this->mailer->getTo());
+    }
+
+    /**
+     * test that default profile is used by constructor if available.
+     *
+     * @return void
+     */
+    public function testDefaultProfile()
+    {
+        $config = ['to' => 'foo@bar.com', 'from' => 'from@bar.com'];
+
+        Configure::write('Mailer.default', $config);
+        Mailer::setConfig(Configure::consume('Mailer'));
```

Comment:
Yes, but to avoid BC breakage I have added Email::setConfig()/getConfig() which proxies calls to Mailer::setConfig()/getConfig(). Same for all methods provided by StaticConfigTrait.